### PR TITLE
Alpha fixed

### PR DESCRIPTION
**Pull request by Minkov**

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-tasks-navigation/ContestTasksNavigation.module.scss [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
-  align-items: start;
+  align-items: flex-start;
   padding: 20px;
   display: flex;
 
   .taskSideNavigationItem {
     text-align: start;
     font-size: 18px;
-    font-family: 'Lato', serif;
+    font-family: Lato, serif;
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-tasks-navigation/ContestTasksNavigation.module.scss:15]: We can export this into a global variable and replace it everywhere

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx [hunk 2]
```diff
@@ -53,25 +54,27 @@ const ContestFilters = ({ onFilterClick }: IContestFiltersProps) => {
...
+                ? styles.btnSelectFilter
+                : '';
 
             return (
                 <Button
-                  type={type}
+                  type={buttonType}
                   onClick={() => handleFilterClick(id)}
-                  className={styles.btnSelectFilter}
+                  className={btmClassName}
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx:74]: btn

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx [hunk 3]
```diff
@@ -53,25 +54,27 @@ const ContestFilters = ({ onFilterClick }: IContestFiltersProps) => {
         [ possibleFilters, onFilterClick ],
     );
 
-    const renderFilterItem = useCallback(
-        ({ id, name }: IFilter) => {
+    const getRenderFilterItemFunc = useCallback(
+        (type: FilterType) => ({ id, name }: IFilter) => {
             // TODO: investigate why filters change ids
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx:59]: log this and reference an issue maybe?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx [hunk 4]
```diff
@@ -53,25 +54,27 @@ const ContestFilters = ({ onFilterClick }: IContestFiltersProps) => {
...
                 <Button
-                  type={type}
+                  type={buttonType}
                   onClick={() => handleFilterClick(id)}
-                  className={styles.btnSelectFilter}
+                  className={btmClassName}
                   text={name}
-                  size={size}
+                  size={ButtonSize.small}
+                  // isWide={isWide}
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx:77]: remove this

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/execution-type-selector/ExecutionTypeSelector.tsx [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import * as React from 'react';
-import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import { get } from 'lodash';
 import styles from './ExecutionTypeSelector.module.scss';
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/execution-type-selector/ExecutionTypeSelector.tsx:4]: styles should be after all imports

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.module.scss [hunk 6]
```diff
@@ -1,21 +1,20 @@
 @import "src/styles/colors";
 
 .contestMainWrapper {
+  // no body
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.module.scss:4]: hmm

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/countdown/Countdown.tsx [hunk 7]
```diff
@@ -26,6 +26,7 @@ interface ICountdownProps {
 
 const defaultRender = (remainingTime: ICountdownRemainingType) => {
     const { hours, minutes, seconds } = convertToTwoDigitValues(remainingTime);
+
     return (
         <>
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/countdown/Countdown.tsx:31]: why is it nesessary to have an empty <> element here?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/headings/Heading.module.scss [hunk 8]
```diff
@@ -28,5 +27,6 @@
   }
 
   &.small {
+    // to be  added
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/headings/Heading.module.scss:30]: maybe mark this as a TODO just to be easier to find TODOs at some point

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/icons/DetailsIconFake.tsx [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+const DetailsIconFake = () => (<>It works</>);
+
+export {
+    // eslint-disable-next-line import/prefer-default-export
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/guidelines/icons/DetailsIconFake.tsx:6]: disable? 😠

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-header/HomeHeader.tsx [hunk 10]
```diff
@@ -1,40 +1,104 @@
-import * as React from 'react';
-
+import React, { FC, useCallback, useEffect, useMemo } from 'react';
+import { isNil } from 'lodash';
+import { useHomeStatistics } from '../../hooks/use-home-statistics';
+import List, { Orientation } from '../guidelines/lists/List';
+import { IDictionary, IKeyValuePair } from '../../common/common-types';
+import StatisticBox from '../statistic-box/StatisticBox';
 import styles from './HomeHeader.module.scss';
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-header/HomeHeader.tsx:7]: styles on bottom

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-header/HomeHeader.tsx [hunk 11]
```diff
@@ -1,40 +1,104 @@
...
+    submissionsCount: 'Submissions',
+    submissionsPerDayCount: 'Submissions per day',
+    problemsCount: 'Problems',
+    strategiesCount: 'Test strategies',
+    contestsCount: 'Contests',
+};
+
+const defeaultProps = { className: styles.icon };
+
+/* eslint-disable react/jsx-props-no-spreading */
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-header/HomeHeader.tsx:29]: disable 😠

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-header/HomeHeader.tsx [hunk 12]
```diff
@@ -1,40 +1,104 @@
...
+    };
 
-    const videoId = window.Keys.YOUTUBE_VIDEO_ID;
+    const renderStatistic = useCallback(
+        (statisticItem: IKeyValuePair<number>) => {
+            const { key, value } = statisticItem;
+
+            return (
+                <StatisticBox
+                  // size={StatisticBoxSize.big}
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-header/HomeHeader.tsx:68]: delete?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-resources/ProblemResources.tsx [hunk 13]
```diff
@@ -1,28 +1,35 @@
 import * as React from 'react';
+import { useCallback } from 'react';
+import { isNil } from 'lodash';
 import { IProblemResourceType } from '../../../common/types';
 import List from '../../guidelines/lists/List';
 import ProblemResource from '../problem-resource/ProblemResource';
 import styles from './ProblemResources.module.scss';
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-resources/ProblemResources.tsx:7]: on bottom

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-resources/ProblemResources.tsx [hunk 14]
```diff
@@ -1,28 +1,35 @@
...
 import concatClassNames from '../../../utils/class-names';
+import { useProblems } from '../../../hooks/use-problems';
 
-interface IProblemResourcesProps {
-    resources: IProblemResourceType[] | undefined
-}
+const ProblemResources = () => {
+    const { state: { currentProblem } } = useProblems();
+
+    const { resources } = currentProblem || {};
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-resources/ProblemResources.tsx:14]: empty line?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-submissions/SubmissionResults.module.scss [hunk 15]
```diff
@@ -9,14 +8,13 @@
...
     background: none;
     cursor: pointer;
   }
 }
 
-.submissionItem {
-  border-bottom: 1px solid $bottom-border-color;
-  padding: 15px 0;
+.submissionsList {
+  // height: 40vh;
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-submissions/SubmissionResults.module.scss:19]: hm?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/statistic-box/StatisticBox.tsx [hunk 16]
```diff
@@ -0,0 +1,71 @@
+import React, { ReactNode, useCallback, useMemo } from 'react';
+import { isNil } from 'lodash';
+import { IStatistic } from '../../common/statistics-types';
+
+import styles from './StatisticBox.module.scss';
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/statistic-box/StatisticBox.tsx:5]: https://www.youtube.com/watch?v=OpWOboXBRB4
(on bottom)

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.module.scss [hunk 17]
```diff
@@ -1,93 +1,46 @@
...
-  color: $primary-red;
-}
+// .correctTestRunHeading {
+//  color: $primary-green;
+// }
 
-.correctTestRunHeading {
-  color: $primary-green;
-}
+// .submissionDetails {
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.module.scss:38]: delete those?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-result-points-label/SubmissionResultPointsLabel.module.scss [hunk 18]
```diff
@@ -1,5 +1,5 @@
 .resultLabel {
-  position: absolute;
-  top: 10px;
-  left: 0;
+  // position: absolute;
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-result-points-label/SubmissionResultPointsLabel.module.scss:2]: delete this and maybe put //something if thats done for silencing eslint

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/SubmissionsList.tsx [hunk 19]
```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { useCallback, useMemo } from 'react';
+import { ISubmissionDetails } from '../../../hooks/submissions/types';
+import concatClassNames from '../../../utils/class-names';
+import { ButtonSize, ButtonState, LinkButton, LinkButtonType } from '../../guidelines/buttons/Button';
+import SubmissionResultPointsLabel from '../submission-result-points-label/SubmissionResultPointsLabel';
+import { formatDate } from '../../../utils/dates';
+import List, { ListType, Orientation } from '../../guidelines/lists/List';
+
+import styles from './SubmissionsList.module.scss';
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/SubmissionsList.tsx:10]: on bottom

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.module.scss [hunk 20]
```diff
@@ -27,14 +31,14 @@
   align-items: center;
 }
 
-.collapsibleButton {
-  font-family: 'Lato', serif;
-  color: $text-dark-gray-color;
-  align-self: flex-end;
-}
+// .collapsibleButton {
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.module.scss:34]: delete?

File: Servers/UI/OJS.Servers.Ui/OJS.Servers.Ui.csproj [hunk 21]
```diff
@@ -43,6 +44,51 @@
       <_ContentIncludedByDefault Remove="Views\Shared\_ValidationScriptsPartial.cshtml" />
       <_ContentIncludedByDefault Remove="ClientApp\dist\9571ccf8bb84a418b2ad.hot-update.json" />
       <_ContentIncludedByDefault Remove="ClientApp\dist\asset-manifest.json" />
+      <_ContentIncludedByDefault Remove="ClientApp\dist\04e46b430b5d6d596c9b.hot-update.json" />
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/OJS.Servers.Ui.csproj:47]: delete

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-tasks-navigation/ContestTasksNavigation.module.scss (hunk 1 above)
Comment (Minkov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-tasks-navigation/ContestTasksNavigation.module.scss:15]: https://github.com/SoftUni-Internal/exam-systems-issues/issues/225

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx (hunk 3 above)
Comment (Minkov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contests-filters/ContestFilters.tsx:59]: The comment is wrong. It should be deleted.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-problem-details/ContestProblemDetails.tsx [hunk 22]
```diff
@@ -1,28 +1,29 @@
...
+import ProblemSubmissions from '../../problems/problem-submissions/ProblemSubmissions';
 
 const ContestProblemDetails = () => {
-    // TODO: remove this. Use `useProblems` inside `ProblemResources`
-    const { state: { currentProblem } } = useProblems();
     const contestTabControlsClass = 'contestTabControls';
-    const contestTabControlsClassName = concatClassNames(styles.contestTabControls, contestTabControlsClass);
-    const contestTabsClassName = 'contestTabs';
+    const problemDetailsContainer = concatClassNames(styles.problemDetailsContainer, contestTabControlsClass);
+    // const contestTabsClassName = 'contestTabs';
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-problem-details/ContestProblemDetails.tsx:11]: delete?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/statistic-box/StatisticBox.tsx [hunk 23]
```diff
@@ -0,0 +1,75 @@
+import React, { ReactNode, useCallback, useMemo } from 'react';
+
+import { isNil } from 'lodash';
+
+import { format } from '../../utils/number-utils';
+
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/statistic-box/StatisticBox.tsx:6]: is a new line between each import needed

File: docker-compose.yml [hunk 24]
```diff
@@ -5,7 +5,7 @@ services:
     build:
       context: .
       dockerfile: Servers/UI/OJS.Servers.Ui/Dockerfile
-    image: ui
```
Comment (shsimeonova) [inline docker-compose.yml:8]: hm

**Review by shsimeonova — APPROVED** [review]

**Pull request merged**